core/src/types/uint.rs

```diff
@@ -27,6 +24,62 @@ construct_uint! {
     pub struct Uint(4);
 }
 
+impl serde::Serialize for Uint {
+    fn serialize<S>(
+        &self,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let amount_string = self.to_string();
+        serde::Serialize::serialize(&amount_string, serializer)
+    }
+}
+
+impl<'de> serde::Deserialize<'de> for Uint {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::Error as serdeError;
+        let amount_string: String =
+            serde::Deserialize::deserialize(deserializer)?;
+
+        let digits = amount_string
+            .chars()
+            .filter_map(|c| {
+                if c.is_numeric() {
```

Comment:
```suggestion
                if c.is_digit(10) {
```
there are non-decimal numberic chars